chains/evm/contracts/test/onRamp/CCVProxy/CCVProxy.parseExtraArgsWithDefaults.t.sol

```diff
@@ -0,0 +1,224 @@
+
+    Client.CCV[] memory requiredCCVs = new Client.CCV[](1);
+    requiredCCVs[0] = Client.CCV({ccvAddress: s_userRequiredCCV1, args: "test args"});
+
+    Client.EVMExtraArgsV3 memory extraArgs = Client.EVMExtraArgsV3({
+      requiredCCV: requiredCCVs,
+      optionalCCV: new Client.CCV[](0),
+      optionalThreshold: 0,
+      finalityConfig: 10,
+      executor: s_userExecutor,
+      executorArgs: "executor args",
+      tokenArgs: "token args"
+    });
+
+    bytes memory encodedExtraArgs = abi.encodePacked(Client.GENERIC_EXTRA_ARGS_V3_TAG, abi.encode(extraArgs));
+
+    Client.EVMExtraArgsV3 memory result =
+      s_ccvProxyTestHelper.parseExtraArgsWithDefaults(destChainConfig, encodedExtraArgs);
+
+    assertEq(result.requiredCCV.length, 1); // Only the original required CCV
+    assertEq(result.requiredCCV[0].ccvAddress, s_userRequiredCCV1);
+    assertEq(result.requiredCCV[0].args, "test args");
+  }
+
+  function test_parseExtraArgsWithDefaults_WithGenericExtraArgsV3Tag_NoCCVsSpecified() public {
+    CCVProxy.DestChainConfig memory destChainConfig = s_defaultDestChainConfig;
+
+    Client.EVMExtraArgsV3 memory extraArgs = Client.EVMExtraArgsV3({
+      requiredCCV: new Client.CCV[](0),
+      optionalCCV: new Client.CCV[](0),
+      optionalThreshold: 0,
+      finalityConfig: 10,
+      executor: address(0),
+      executorArgs: "",
+      tokenArgs: ""
+    });
+
+    bytes memory encodedExtraArgs = abi.encodePacked(Client.GENERIC_EXTRA_ARGS_V3_TAG, abi.encode(extraArgs));
+
+    Client.EVMExtraArgsV3 memory result =
+      s_ccvProxyTestHelper.parseExtraArgsWithDefaults(destChainConfig, encodedExtraArgs);
+
+    assertEq(result.requiredCCV.length, 2); // 1 from config + 1 default
+    assertEq(result.requiredCCV[0].ccvAddress, REQUIRED_CCV); // Required CCV from config
+    assertEq(result.requiredCCV[1].ccvAddress, DEFAULT_CCV); // Default CCV
+    assertEq(result.executor, DEFAULT_EXECUTOR); // Default executor
+  }
+
+  function test_parseExtraArgsWithDefaults_WithLegacyExtraArgs() public {
+    CCVProxy.DestChainConfig memory destChainConfig = s_defaultDestChainConfig;
+
+    bytes memory legacyExtraArgs = "legacy args";
+
+    Client.EVMExtraArgsV3 memory result =
+      s_ccvProxyTestHelper.parseExtraArgsWithDefaults(destChainConfig, legacyExtraArgs);
+
+    assertEq(result.requiredCCV.length, 2); // 1 from config + 1 default
+    assertEq(result.requiredCCV[0].ccvAddress, REQUIRED_CCV); // Required CCV from config
+    assertEq(result.requiredCCV[1].ccvAddress, DEFAULT_CCV); // Default CCV with legacy args
+    assertEq(result.requiredCCV[1].args, legacyExtraArgs);
+    assertEq(result.executorArgs, legacyExtraArgs);
+    assertEq(result.executor, DEFAULT_EXECUTOR); // Default executor
+  }
+
+  function test_parseExtraArgsWithDefaults_RevertWhen_InvalidOptionalCCVThreshold() public {
+    CCVProxy.DestChainConfig memory destChainConfig = CCVProxy.DestChainConfig({
```

Comment:
Maybe this can go to the setup func and you can override only when needed in tests?